lib/types/object/index.js

```diff
@@ -118,18 +119,53 @@ internals.Object = class extends Any {
                 }
             }
 
-            if (target[rename.from] === undefined) {
-                delete target[rename.to];
+            if (rename.isRegExp) {
```

Comment:
The place of this `if` is likely wrong, it should be a regexp path or a string path, trying to use both is madness. For example you were just lucky in your `ignoreUndefined` test cases :
```js
const schema = Joi.object({
    b: Joi.any()
}).rename(/^a$/, 'b', { ignoreUndefined: true });

schema.validate({ a: 1 }) // Errors because it failed to rename a to b, it stopped the loop in the 1st statement
```

Reply:
Hey alright so we addressed that test case, but I honestly have mixed feelings on the implementation for the fix. We are using a `.filter` to do the undefined check and also prevent a duplicate future iteration as a side effect of the `filter` callback. Don't like that, but it does prevent double iteration over the same keys.

Let us know what you think of this approach. We can also just split the regexp and string paths completely; that just meant duplicating the exact logic for the `multiple` and `override` checks into both blocks.